contribs/gnofaucet/gh.go

```diff
@@ -87,12 +164,64 @@ func gitHubClaimMiddleware(coolDownLimiter cooldownLimiter) faucet.Middleware {
 				)
 			}
 
-			// Make sure the method is "drip"
+			if len(req.Params) != 1 {
+				return spec.NewJSONResponse(
+					req.ID,
+					nil,
+					spec.NewJSONError("params must contain only the address", spec.InvalidParamsErrorCode),
+				)
+			}
+			reward, err := rewarder.GetReward(ctx, username)
+			if err != nil {
+				return spec.NewJSONResponse(
+					req.ID,
+					nil,
+					spec.NewJSONError(fmt.Sprintf("unable to get reward: %v", err), spec.ServerErrorCode),
+				)
+			}
+
+			if reward == 0 {
+				return spec.NewJSONResponse(
+					req.ID,
+					nil,
+					spec.NewJSONError("no tokens to reward.", spec.ServerErrorCode),
+				)
+			}
+
+			req.Method = faucet.DefaultDripMethod
+
+			c := std.NewCoin("ugnot", int64(reward)).String()
+			req.Params = append(req.Params, c)
+
+			// TODO: this should be called AFTER the faucet succesfully gives tokens to the requester
```

Comment:
I'm trying to think of an elegant way to handle this

If we know the `gitHubClaimMiddleware` is the last middleware to run, what do you think about calling something like `next.ServeHTTP` instead of `next(...)` and just applying the reward there? I'd need to add support for this in the `gnolang/faucet`

I think `gitHubClaimRewardsMiddleware` and `gitHubClaimMiddleware` should be combined into one, this way you also don't have to alter the request itself

Reply:
@gfanton what do you think?